### ISSUE TITLE
Why there is a hidden activation loss in harmful loss?

### ISSUE THREAD

**Issue by huangtiansheng** [domenicrosati/representation-noising#2]
I am also wondering why several hidden activation loss is added to the harmful loss? My understanding is that you add a classifier upon the hidden activation (but not only in the final embeding) and do cross entropy for next word prediction.  Please correct me if that is not your intention...

```
 for i, h in enumerate(harmful_outputs.hidden_states):
      # print(h.dtype)
      h = h.to(DEVICE).half()
      out = output_embeddings(norm(h))
      loss = masked_token_ce_loss(
          out.to(DEVICE), harmful_batch['input_ids'].to(DEVICE),
          mask
      )
      harmful_losses += loss
```

**Comment by domenicrosati** [comment]
This corresponds to lines 7-11 in Algorithmn 1 Appendix B1.

Where we say:
> For our layer-wise gradient ascent losses, we use a
similar approach to the tuned lens [39] and use the language model head on top of each activation to
compute language modelling loss at each layer

We do this to minimize as much information across layers as possible.

I am realizing now we don't have a proper ablation in the paper showing what happens if you only do this at the final layer but from our previous experiments it is much worse. I will see if we can add these results to the paper.
